main.go

```diff
@@ -332,13 +336,75 @@ Note: may not work with some transports such as p2p-circuit (not applicable) and
 							}
 							return vole.Ping(c.Context, c.Bool("force-relay"), ai)
 						},
+					}, {
+						Name:      "http",
+						ArgsUsage: "<multiaddr>",
+						Flags: []cli.Flag{
+							&cli.StringFlag{
+								Name:        "socket-path",
+								Usage:       `Use the specified path for the unix socket instead of making a new one.`,
+								DefaultText: "",
+								Value:       "",
+							},
+						},
+						Usage: "Make http requests to the given multiaddr with a unix socket",
```

Comment:
Seems good enough for now, certainly for more in-depth debugging of a specific endpoint. Wonder if supporting some basic curl like functionality would be good here so people can do one-line http-over-libp2p requests. WDYT?

Reply:
maybe a simple GET request, and anything else would require the unix socket proxy? Does the unix socket proxy trick even work on windows?

Implementing the GET request sounds good to me, but how about we do that in a future PR once the next go-libp2p  version is released? That one includes code that will make this easier. User's could make a request to `multiaddr:/dns4/example.com/http/http-path/subpath` and we can just pass that to go-libp2p.